server/handlers/converters.go

```diff
@@ -97,8 +98,12 @@ func (ah *ActivityHandler) handleMentions(msg *clientmodels.Message) string {
 		mmMention := ""
 		mention := msg.Mentions[idx]
 		idx++
-		switch {
-		case mention.UserID != "":
+		if mention.UserID != "" {
+			if userOrConversationIDVsNames[mention.UserID] != mention.MentionedText {
+				msg.Text = strings.Replace(msg.Text, fmt.Sprintf("&nbsp;<at id=\"%s\">%s</at>", fmt.Sprint(mention.ID), mention.MentionedText), "", 1)
```

Comment:
why are we also replacing the initial space here?

Reply:
The mention `Dummy User` on Teams should be converted to `@dummyuser` on Mattermost without a trailing space. This is done to remove that trailing space.